CONTRIBUTING.md

```diff
@@ -65,3 +54,53 @@ For new features that affect test coverage, use a `coverage/*` branch in this re
+
+How to run them both locally:
+
+* **Reference Implementation**: To run this locally, follow the guidelines in its [project's contribution documentation](https://github.com/openactive/OpenActive.Server.NET/blob/master/CONTRIBUTING.md).
+* **Test Suite**: With the reference implementation running locally, use the [`default-dev.json` config](./config/default-dev.json) by setting env var `NODE_APP_INSTANCE=dev`.
+
+    For example, to run broker and a specific test separately (See [**Tips for quicker test runs**](#tips-for-quicker-test-runs) for a more efficient process):
+
+    ```sh
+    NODE_APP_INSTANCE=dev npm start
+    ```
+
+**N.B.:** When running the reference implementation locally, the `unknown-endpoint` test in `core/common-error-conditions` will fail as the dev version of reference implementation shows a developer error page instead of the regular 404 response.
+
+### Tips for quicker test runs
+
+Running all of the tests can take a long time. To speed up your development/testing feedback loop, you can:
+
+1. Run a single test at a time, rather than running all of the tests.
+2. Keep [Broker Microservice](./packages/openactive-broker-microservice/) running in the background, rather than restarting it for each test.
+
+Here's an example of how to do this:
+
+```sh
+# In one terminal, start broker
+NODE_APP_INSTANCE=dev npm run start-broker
+
+# Then, in ANOTHER terminal, run the opportunity-in-past single test
+NODE_APP_INSTANCE=dev npm run start-tests -- packages/openactive-integration-tests/test/features/core/common-error-conditions/implemented/opportunity-in-past-test.js
+# OR, run only the common-error-conditions feature
+NODE_APP_INSTANCE=dev npm run start-tests -- packages/openactive-integration-tests/test/features/core/common-error-conditions/
+```
+
+#### Even quicker test runs - reduce random test data
+
+When using the [`default-dev.json` config](./config/default-dev.json) (i.e. with `NODE_APP_INSTANCE=dev`), Test Suite is set to use [**controlled mode**](https://developer.openactive.io/open-booking-api/key-decisions#controlled-mode). This means that it creates all the data that it needs for testing. Therefore, the [reference implementation](#reference-implementation) does not need to generate its own data, which it does by default.
+
+Turning this off will speed up the bootstrapping of both reference implementation and broker microservice, which no longer needs to harvest lots of data.
+
+To do this, see the [**Optimizing for Controlled Mode**](https://github.com/openactive/OpenActive.Server.NET/blob/master/CONTRIBUTING.md#optimizing-for-controlled-mode) section of OpenActive.Server.NET's contribution documentation.
```

Comment:
links to a section created in this PR: https://github.com/openactive/OpenActive.Server.NET/pull/216